cpp_exercise/src/lib/ipcmsgqlib.cc

```diff
@@ -0,0 +1,166 @@
+#include "ipcmsgqlib.h"
+#include "ipcexceptionlib.h"
+
+
+/***
+ * 
+ * MsgQueueSender Class
+ * 
+ * 
+ * **/
+
+MsgQueueSender::MsgQueueSender(const std::string & filename): _file_name(filename){
+        std::cout<<"**************************************************************"<<std::endl;
+        std::cout<<"****************************IPCSender: MESSAGE QUEUE PROTOCOL**************************"<<std::endl;
+        std::cout<<"**************************************************************"<<std::endl;
+        std::cout<<"**************************************************************"<<std::endl;
+
+   
+
+    int err;
+    remove(_msg_queue_name.c_str());
+        ret = mq_unlink(_msg_queue_name.c_str());
+
+    
+     memset(&_attrs, 0, sizeof _attrs);
+   
+
+    _attrs.mq_maxmsg = _msg_queue_maxnummsg;
+    _attrs.mq_msgsize = _msg_queue_msgsize;
+
+   FileHandler fd(_file_name);
+    
+    std::cout<<"          File Name: "<<_file_name<<std::endl;
+    std::cout<<"          File Size: "<<fd.getSize()<<" bytes"<<std::endl;
+        std::cout<<"          Loading the File. Please Wait..."<<std::endl;
+
+
+    
+
+
+    _msg_queue = mq_open(_msg_queue_name.c_str(), O_WRONLY | O_CREAT , S_IRWXU | S_IRWXG, &_attrs);
+     if(_msg_queue<0){
+         std::cout<<strerror(errno)<<std::endl;
+         throw IPCException("IPCSender ERROR: Cannot open the MSGQueue.");
+
+    }
+   
+ std::cout<<"          MSGQueue is Open: Trying to send the file  "<<std::endl;
+ 
+ fd.openForReading();
```

Comment:
If openForReading() throws, the destructor of MsgQueueSender won't be called (only destructors of objects completely created are called), so the queue will stay opened/linked.

Reply:
I changed the position of this line to upper but I think I should bring back the try and catch for file handling in my ipc protocols

Reply:
Ook for this method, but the issue is the same for readFile() (line 61)

Reply:
Yes you are right. Since I have the same issue in my other protocols I should consider this as an issue and will resolve it restructuring my code and adding a new function for all of transfering instead of having them all in constructor